sensor/kubernetes/crs/crs.go

```diff
@@ -0,0 +1,299 @@
+	}
+	k8sClient := k8sutil.MustCreateK8sClient(config)
+
+	centralHello, err := centralHandshake(ctx, k8sClient, centralConnection)
+	if err != nil {
+		return errors.Wrap(err, "handshake with central")
+	}
+
+	// Store certificates+keys contained in Central's centralHello response as
+	// Kubernetes secrets named `tls-cert-<service slug name>`.
+	err = persistCertificates(ctx, centralHello.GetCertBundle(), k8sClient)
+	if err != nil {
+		return errors.Wrap(err, "persisting certificates")
+	}
+
+	return nil
+}
+
+// temporarilyStoreRegistrantSecret extracts the REGISTRANT_SERVICE certificate+key pair from the CRS
+// and stores it alongside the CA certificate in the temporary storage /run/secrets/stackrox.io/certs.
+func temporarilyStoreRegistrantSecret(crs *crs.CRS) error {
+	// Extract (first) CA certificate from the CRS.
+	var caCert string
+	if len(crs.CAs) > 0 {
+		caCert = crs.CAs[0]
+	}
+	if caCert == "" {
+		return errors.New("malformed Cluster Registration Secret (missing CA certificate)")
+	}
+
+	for fileName, content := range map[string]string{
+		"ca.pem":   caCert,
+		"cert.pem": crs.Cert,
+		"key.pem":  crs.Key,
+	} {
+		filePath := filepath.Join(mtls.CertsPrefix, fileName)
+		if err := os.WriteFile(filePath, []byte(content), 0600); err != nil {
+			return errors.Wrapf(err, "writing mTLS material to file %q", filePath)
+		}
+		log.Infof("Successfully wrote file %s.", filePath)
+	}
+
+	return nil
+}
+
+func openCentralConnection() (*grpcUtil.LazyClientConn, error) {
+	// Create central client.
+	centralEndpoint := env.CentralEndpoint.Setting()
+	centralClient, err := centralclient.NewClient(centralEndpoint)
+	if err != nil {
+		return nil, errors.Wrapf(err, "initializing Central client for endpoint %s", env.CentralEndpoint.Setting())
+	}
+
+	centralConnFactory := centralclient.NewCentralConnectionFactory(centralClient)
+	centralConnection := grpcUtil.NewLazyClientConn()
+	certLoader := centralclient.RemoteCertLoader(centralClient)
+	go centralConnFactory.SetCentralConnectionWithRetries(centralConnection, certLoader)
+
+	log.Infof("Connecting to Central server %s", centralEndpoint)
+
+	// Wait for connection to be ready.
+	okSig := centralConnFactory.OkSignal()
+	errSig := centralConnFactory.StopSignal()
+	select {
+	case <-errSig.Done():
+		return nil, errors.Wrap(errSig.Err(), "waiting for Central connection from factory")
+	case <-okSig.Done():
+	}
+
+	log.Info("Central connection ready.")
+	return centralConnection, nil
+}
+
+// centralHandshake performs the hello-handshake with Central and returns Central's CentralHello reponse on success.
+func centralHandshake(ctx context.Context, k8sClient kubernetes.Interface, centralConnection *grpcUtil.LazyClientConn) (*central.CentralHello, error) {
+	sensorHello, err := prepareSensorHelloMessage(k8sClient)
+	if err != nil {
+		return nil, errors.Wrap(err, "preparing SensorHello message")
+	}
+
+	ctx = metadata.AppendToOutgoingContext(ctx, centralsensor.SensorHelloMetadataKey, "true")
```

Comment:
This `true` seems like a magic constant. Should we perhaps extract this to a function that the regular sensor also uses?

Reply:
Not sure what the best approach would be for this. This `"true"` is currently used in a couple of places. Ok, to keep this out-of-scope for this PR?

Reply:
I think it depends. If the pre-existing uses are close together (like in one file or at least one package), then I'd insist on tackling this before this use sprawls to this new package.
If OTOH they are already distributed here and there, then we're not making things significantly worse here.

Reply:
Looks like this is already in 4 different files across 3 packages. So I guess it's OK to keep this out of scope of this PR.
![image](https://github.com/user-attachments/assets/a1f458b7-58b1-4d6c-964e-89263aa52738)